sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/pipeline/TelemetryItemExporter.java

```diff
@@ -129,6 +132,40 @@ CompletableResultCode internalSendByBatch(TelemetryItemBatchKey telemetryItemBat
         return telemetryPipeline.send(byteBuffers, telemetryItemBatchKey.connectionString, listener);
     }
 
+    // serialize an array of TelemetryItems to an array of byte buffers
+    private static List<ByteBuffer> serialize(List<TelemetryItem> telemetryItems) {
+        try {
+            ByteBufferOutputStream out = writeTelemetryItemsAsByteBufferOutputStream(telemetryItems);
+            out.close(); // closing ByteBufferOutputStream is a no-op, but this line makes LGTM happy
+            List<ByteBuffer> byteBuffers = out.getByteBuffers();
+            for (ByteBuffer byteBuffer : byteBuffers) {
+                byteBuffer.flip();
+            }
+            return out.getByteBuffers();
+        } catch (IOException e) {
+            throw new IllegalStateException("Failed to serialize list of TelemetryItems to List<ByteBuffer>", e);
+        }
+    }
+
+    // gzip and add new line delimiter from a list of telemetry items to a byte buffer output stream
+    private static ByteBufferOutputStream writeTelemetryItemsAsByteBufferOutputStream(List<TelemetryItem> telemetryItems) throws IOException {
+        try (ByteBufferOutputStream result = new ByteBufferOutputStream(new AppInsightsByteBufferPool())) {
```

Comment:
let's cache the pool

although I just noticed that it looks like we stopped putting used byte buffers back into the pool a long time ago (tracked down to https://github.com/microsoft/ApplicationInsights-Java/pull/2091), which means we aren't getting any benefit from pooling, oops